### PR TITLE
scripts: compliance: More cleanups

### PR DESCRIPTION
**Pull request by carlescufi**
More cleanups in preparation for some extra annotation work.

### CODE REVIEW

File: scripts/ci/check_compliance.py [hunk 1]
```diff
@@ -82,9 +82,10 @@ def __init__(self, severity, title, file, line, col=None, desc=""):
         description = f':{desc}' if desc else ''
         msg_body = desc or title
 
-        txt = f'\n{title}{description}\nFile:{file}\nLine:{line}' + \
+        txt = f'\n{title}{description}\nFile:{file}' + \
+              (f'\nLine:{line}' if line else '') + \
```
Comment (gmarull) [inline scripts/ci/check_compliance.py:75]: suggestion: `line or ‘’`

**Review by gmarull — DISMISSED** [review]

File: scripts/ci/check_compliance.py (hunk 1 above)
Comment (carlescufi) [reply scripts/ci/check_compliance.py:75]: No, actually that won't work because the string is `\nLine:{line}`

**Comment by carlescufi** [comment]
@gmarull couldn't apply your suggestion but added an extra commit

**Review by gmarull — DISMISSED** [review]

File: scripts/ci/check_compliance.py [hunk 2]
```diff
@@ -566,98 +579,81 @@ def check_no_undef_outside_kconfig(self, kconf):
...
 
+    # Many of these are symbols used as examples. Note that the list is sorted
+    # alphabetically, and skips the CONFIG_ prefix.
+    UNDEF_KCONFIG_WHITELIST = {
+        "ALSO_MISSING",
+        "APP_LINK_WITH_",
+        "APP_LOG_LEVEL", # Application log level is not detected correctly as
+                         # the option is defined using a template, so it can't
+                         # be grepped
+        "ARMCLANG_STD_LIBC",  # The ARMCLANG_STD_LIBC is defined in the
```
Comment (stephanosio) [inline scripts/ci/check_compliance.py:590]: > scripts/ci/check_compliance.py:590 Trailing whitespace (trailing-whitespace)

check_compliance seems to be very effective against itself :)

Comment (carlescufi) [reply scripts/ci/check_compliance.py:590]: 🤣

**Review by gmarull — APPROVED** [review]

**Review by stephanosio — APPROVED** [review]

**Comment by nashif** [comment]
post merge +1 :)

**Pull request merged**